From 92d90d1c421d8c0b7adcf75d91592d5e77946c01 Mon Sep 17 00:00:00 2001
From: Tobias <ruck.tobias@gmail.com>
Date: Tue, 26 Nov 2024 19:32:47 +0100
Subject: [PATCH] [doged] Fix `sendtoaddress` (#87)

There was an oversight in `CWallet::SignTransaction`, which is fixed in this PR.

diff --git a/src/wallet/wallet.cpp b/src/wallet/wallet.cpp
index 81fb31cf276f4c8f15bf570c8cc4a0850e623a6f..ca7714f1ecc8ddcbd56612ee21d789b3b5f5eb68 100644
--- a/src/wallet/wallet.cpp
+++ b/src/wallet/wallet.cpp
@@ -2106,8 +2106,8 @@ bool CWallet::SignTransaction(CMutableTransaction &tx) const {
     std::map<int, std::string> input_errors;
     return SignTransaction(tx, coins,
                            GetSignScriptFlags() & SCRIPT_ENABLE_SIGHASH_FORKID
-                               ? SigHashType()
-                               : SigHashType().withForkId(),
+                               ? SigHashType().withForkId()
+                               : SigHashType(),
                            input_errors);
 }
 
